sharding-core/sharding-core-preprocessor/src/test/java/org/apache/shardingsphere/core/optimize/segment/insert/InsertValueContextTest.java

```diff
@@ -0,0 +1,155 @@
+
+import com.google.common.collect.Lists;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.apache.shardingsphere.core.optimize.segment.insert.expression.DerivedLiteralExpressionSegment;
+import org.apache.shardingsphere.core.optimize.segment.insert.expression.DerivedParameterMarkerExpressionSegment;
+import org.apache.shardingsphere.core.parse.sql.segment.dml.expr.ExpressionSegment;
+import org.apache.shardingsphere.core.parse.sql.segment.dml.expr.simple.LiteralExpressionSegment;
+import org.apache.shardingsphere.core.parse.sql.segment.dml.expr.simple.ParameterMarkerExpressionSegment;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+public class InsertValueContextTest {
+
+    @Test
+    public void assertInstanceConstructedOk() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        Collection<ExpressionSegment> assignments = Lists.newArrayList();
+        List<Object> parameters = Lists.newArrayList();
+        int parametersOffset = 0;
+
+        InsertValueContext insertValueContext = new InsertValueContext(assignments,parameters, parametersOffset);
+
+        Method calculateParametersCountMethod = InsertValueContext.class.getDeclaredMethod("calculateParametersCount", Collection.class);
+        calculateParametersCountMethod.setAccessible(true);
+        int calculateParametersCountResult = (int) calculateParametersCountMethod.invoke(insertValueContext, new Object[] {assignments});
+        assertThat(insertValueContext.getParametersCount(), is(calculateParametersCountResult));
+
+        Method getValueExpressionsMethod = InsertValueContext.class.getDeclaredMethod("getValueExpressions", Collection.class);
+        getValueExpressionsMethod.setAccessible(true);
+        List<ExpressionSegment> getValueExpressionsResult = (List<ExpressionSegment>) getValueExpressionsMethod.invoke(insertValueContext, new Object[] {assignments});
+        assertThat(insertValueContext.getValueExpressions(), is(getValueExpressionsResult));
+
+        Method getParametersMethod = InsertValueContext.class.getDeclaredMethod("getParameters", new Class[]{List.class, int.class});
+        getParametersMethod.setAccessible(true);
+        List<Object> getParametersResult = (List<Object>) getParametersMethod.invoke(insertValueContext, new Object[] {parameters, parametersOffset});
+        assertThat(insertValueContext.getParameters(), is(getParametersResult));
+    }
+
+    @Test
+    public void assertGetValueWhenParameterMarker() {
+        Collection assignments = makeParameterMarkerExpresstionSegment();
+        String parameterValue = "test";
+        List parameters = Lists.newArrayList(parameterValue);
```

Comment:
Raw list without generic type is not permit

Reply:
To avoid this problem, it seams that the following constructor should been defined : 

```java 
public InsertValueContext(final Collection<? extends ExpressionSegment> assignments, final List<? extends Object> parameters, final int parametersOffset) {
```

that is, assignments and parameters should use generic subtypes with the format of "? extends XXX". 

If there is another solution, please let know.

Reply:
solved. 
@terrymanu    Thank you  very much for your patient guidance